pkg/utils/checksum.go

```diff
@@ -0,0 +1,19 @@
+package utils
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"os"
+)
+
+// Checksum reads the file at the provided path and calculates the sha256sum
+func Sha256sum(filepath string) (string, error) {
+	fileBytes, err := os.ReadFile(filepath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read file '%s' while generating sha256sum: %w", filepath, err)
+	}
+	sumBytes := sha256.Sum256(fileBytes)
+	// TODO - there's probably a better way to do this
+	return fmt.Sprintf("%x", sumBytes[:]), nil
```

Comment:
I think this is basically the way to do this - https://pkg.go.dev/crypto/sha256#example-Sum256